src/app/seamly2d/mainwindow.cpp

```diff
@@ -53,298 +53,358 @@
-    , groupsWidget(nullptr)
-    , patternPiecesWidget(nullptr)
-    , lock(nullptr)
-    , zoomScaleSpinBox(nullptr)
-    , m_penToolBar(nullptr)
-    , m_penReset(nullptr)
-    , m_zoomToPointComboBox(nullptr)
-{
-    for (int i = 0; i < MaxRecentFiles; ++i)
-    {
-        recentFileActs[i] = nullptr;
-    }
+    
+    : MainWindowsNoGUI(parent)                 
+    , ui(new Ui::MainWindow)                   
+    , watcher(new QFileSystemWatcher(this))    
+    , currentTool(Tool::Arrow)                 
+    , lastUsedTool(Tool::Arrow)                
+    , draftScene(nullptr)                      
+    , pieceScene(nullptr)                      
+    , mouseCoordinates(nullptr)                
+    , infoToolButton(nullptr)                  
+    , helpLabel(nullptr)                       
+    , isInitialized(false)                     
+    , mChanges(false)                          
+    , mChangesAsked(true)                      
+    , patternReadOnly(false)                   
+    , dialogTable(nullptr)                     
+    , dialogTool()                             
+    , historyDialog(nullptr)                   
+    , fontComboBox(nullptr)                    
+    , fontSizeComboBox(nullptr)                
+    , draftBlockComboBox(nullptr)              
+    , draftBlockLabel(nullptr)                 
+    , mode(Draw::Calculation)                  
+    , currentBlockIndex(0)                     
+    , currentToolBoxIndex(0)                   
+    , isToolOptionsDockVisible(true)           
+    , isGroupsDockVisible(true)                
+    , isLayoutsDockVisible(false)              
+    , isToolboxDockVisible(true)               
+    , drawMode(true)                           
+    , recentFileActs()                         
+    , separatorAct(nullptr)                    
+    , leftGoToStage(nullptr)                   
+    , rightGoToStage(nullptr)                  
+    , autoSaveTimer(nullptr)                   
+    , guiEnabled(true)                         
+    , gradationHeights(nullptr)                
+    , gradationSizes(nullptr)                  
+    , gradationHeightsLabel(nullptr)           
+    , gradationSizesLabel(nullptr)             
+    , toolProperties(nullptr)                  
+    , groupsWidget(nullptr)                    
+    , patternPiecesWidget(nullptr)             
+    , lock(nullptr)                            
+    , zoomScaleSpinBox(nullptr)                
+    , m_penToolBar(nullptr)                    
+    , m_penReset(nullptr)                      
+    , m_zoomToPointComboBox(nullptr)           
+
+    // define Seamly2D main window
+    {
+        for (int i = 0; i < MaxRecentFiles; ++i)
+        {
+            recentFileActs[i] = nullptr;           // Initialize recent file actions array
+        }
+
+        createActions();                           // Create actions
+        initializeScenes();                        // Initialize scenes
+
```

Comment:
Same, the code there is human readable. 
I find those unnecessary end-of-line comments counter productive since they add difficulties to code: if I want to add a line between the two I will probably click immediately after the semicolon then press enter, before seeing that I messed up with the comment that is now on the wrong line, with the wrong number of tabs/spaces before, etc.